diff --git a/setup.py b/setup.py
index 267e406..6f82149 100644
--- a/setup.py
+++ b/setup.py
@@ -1,5 +1,15 @@
+from itertools import chain
 from setuptools import setup

+extras = {
+    'moola': ['moola>=0.1.6'],
+    'test': ['pytest'],
+    'doc': ['sphinx', 'sphinxcontrib-bibtex'],
+    'visualisation': ['tensorflow'],
+}
+# 'all' includes all of the above
+extras['all'] = list(chain(*extras.values()))
+
 setup(name='dolfin_adjoint',
       version='2018.1.0',
       description='High-level automatic differentiation library for FEniCS.',
@@ -14,5 +24,6 @@ setup(name='dolfin_adjoint',
                 'pyadjoint.optimization'],
       package_dir={'fenics_adjoint': 'fenics_adjoint', 'pyadjoint': 'pyadjoint',
                    'firedrake_adjoint': 'firedrake_adjoint', 'dolfin_adjoint': 'dolfin_adjoint'},
-      install_requires=['scipy', 'pytest', 'sphinx', 'sphinxcontrib-bibtex', 'moola>=0.1.6', 'tensorflow'],
-      )
+      install_requires=['scipy'],
+      extras_require=extras,
+)
