commit d886230fd1ffc96f08e9babd3f8328798345fcd7
Author: Johannes Ring <johannr@simula.no>
Date:   Mon Jan 23 18:35:21 2017 +0100

    Fix syntax for py3

diff --git a/dolfin_adjoint/timeforms.py b/dolfin_adjoint/timeforms.py
index 768fd7c..5daae4b 100644
--- a/dolfin_adjoint/timeforms.py
+++ b/dolfin_adjoint/timeforms.py
@@ -211,7 +211,7 @@ class TimeMeasure(object):
             interval = slice(START_TIME,FINISH_TIME,None)
 
         if interval not in (START_TIME,FINISH_TIME) and not isinstance(interval, (slice, int, float)):
-            raise ValueError, "TimeMeasure can only be indexed with floats, START_TIME and FINISH_TIME."
+            raise ValueError("TimeMeasure can only be indexed with floats, START_TIME and FINISH_TIME.")
 
         self.interval = timeslice(interval)
 
